src/OrchardCore.Modules/OrchardCore.Tenants/Controllers/AdminController.cs

```diff
@@ -521,16 +506,12 @@ public async Task<IActionResult> Disable(string id)
                 return Forbid();
             }
 
-            var shellSettings = _shellHost.GetAllSettings()
-                .Where(s => String.Equals(s.Name, id, StringComparison.OrdinalIgnoreCase))
-                .FirstOrDefault();
-
-            if (shellSettings == null)
+            if (!_shellHost.TryGetSettings(id, out var shellSettings))
             {
                 return NotFound();
             }
 
-            if (String.Equals(shellSettings.Name, ShellHelper.DefaultShellName, StringComparison.OrdinalIgnoreCase))
+            if (shellSettings.IsDefaultShell())
```

Comment:
Yes as the related extension does the same currently, but not sure I like this extension.

1. To be more strict before doing an action **not allowed on the Default tenant**, we have better to do a case insensitive comparison. So here okay as this is what the extension currently does.

2. But in other places, before doing an action only **allowed by the Default tenant**, I would say that we have better to do a case sensitive comparison, in that case `IsDefaultShell()` is less restrictive and not right for me.

We could remove `IsDefaultShell()` and just use `==` or `String.Equals(...OrdinalIgnoreCase)` as here if it still makes sense ;) but removing it is a breaking change, or just make this extension cases sensitive (see below).

But `IsDefaultShell()` is always used on a registered shellSettings, we can create any non registered shell with any name, but we can't register e.g. a `dEfAuLt` tenant if the tenant validator was used, and moreover with the new case insensitive dicos it would have already failed on a duplicate key.

So anyway I think that's okay if `IsDefaultShell()` is case sensitive or not.

---

So first we could change `IsDefaultShell()` like this as it makes more sense for me, in most of the places that we are using it we are in the above case number `2.`.

    public bool IsDefaultShell() => Name == ShellHelper.DefaultShellName;

Then because it would be case sensitive, revert the usage of `IsDefaultShell()` **here and around line 254/248** where we are in the case number `1.`. OR because of the above analysis, keep this change **here and line 254/248** just because the code is cleaner.

I'm hesitating ;)

**In any case you can use** `IsDefaultShell()` **to init the** `ShellSettingsEntry.IsDefaultTenant` **line 108**, which is checked in the index view and where we are in the case number `2.`. At least it would be as before if we don't change the extension, and better if we change it.

Reply:
I am not sure that I agree, or may be I do but could be missing the point. 

First, `IsDefaultShell()` checks if the settings is a host settings. Since this PR is allowing invariant tenant, then `Default`, `dEfAuLt`, `default` or any other form should be treated the same. So I think, `IsDefaultShell() => String.Equals(Name, ShellHelper.DefaultShellName, StringComparison.OrdinalIgnoreCase)` is valid.

In fact, I think we should add another extension method to `ShellSettings` to clean up some of the other code. I think we should add `IsShell(string tenantName) => String.Equals(Name, tenantName, StringComparison.OrdinalIgnoreCase);`

I don't know that I like the name `IsShell` perhaps you can come up with a better name if you agree. But at the end the combined code will look like this

```
public bool IsDefaultShell() => IsShell(ShellHelper.DefaultShellName);
public bool IsShell(string tenantName) => String.Equals(Name, tenantName, StringComparison.OrdinalIgnoreCase);
```

Now, instead of having to write code like `if (String.Equals(shellSettings.Name, "name", StringComparison.OrdinalIgnoreCase))` we would simplify it to `shellSettings.IsShell("name")`.

Reply:
> Since this PR is allowing invariant tenant, then Default, dEfAuLt, default or any other form should be treated the same.

I don't think so, this PR only allows to retrieve a tenant by name in a case insensitive way, then for example the name of the default registered tenant is still always equal to the const name `ShellHelper.DefaultShellName` which currently is `Default`.

Anyway I think it is better to cleanup this PR, maybe open a new one to only focus on the new case insensitive shell inner dictionaries. I will comment it in the main thread.